client/src/App.js

```diff
@@ -22,7 +23,15 @@ export default class App extends Component {
     return (
       <div>
         <SavedList list={this.state.savedList} />
-        <div>Replace this Div with your Routes</div>
+        <Route exact path="/" component={MovieList} />
+        <Route
```

Comment:
Looks like you have 2 components rendering to the path /movies/:id. that is why you are getting that double movie card whenever you select a movie. Notice how the one on top has a functioning save button while the one on bottom does not. Look at how you defined these 2 routes. The first one renders the component and passes in props. The second one uses a component prop and does not pass in your add to saved list function.